ui/selectors/multichain-accounts/feature-flags.ts

```diff
@@ -0,0 +1,109 @@
+/* eslint-disable @typescript-eslint/no-unused-vars */
+// We can ignore the unused vars warning while the flag is not active
+
+import {
+  getRemoteFeatureFlags,
+  type RemoteFeatureFlagsState,
+} from '../remote-feature-flags';
+
+/**
+ * Feature flag type for multichain accounts features
+ */
+export type MultichainAccountsFeatureFlag = {
+  enabled: boolean;
+  featureVersion: string | null;
+  minimumVersion: string | null;
+};
+
+// TODO: Update the value to the decided version multichain accounts will be released
+const MINIMUM_SUPPORTED_VERSION = null;
+const FEATURE_VERSION_1 = '1';
+const FEATURE_VERSION_2 = '2';
+
+/**
+ * Compare is the version1 is greater than or equal to version2
+ *
+ * @param version1 - The first version string to compare.
+ * @param version2 - The second version string to compare.
+ * @returns boolean - True if version1 is greater than or equal to version2, false otherwise.
+ */
+export const compareVersions = (version1: string, version2: string) => {
+  const regex = /^\d+\.\d+\.\d+$/u;
+  if (!regex.test(version1) || !regex.test(version2)) {
+    return false; // Invalid version format
+  }
+
+  const v1 = version1.split('.').map(Number);
+  const v2 = version2.split('.').map(Number);
+
+  for (let i = 0; i < Math.max(v1.length, v2.length); i++) {
+    const num1 = v1[i] || 0;
+    const num2 = v2[i] || 0;
+
+    if (num1 > num2) {
+      return true;
+    }
+    if (num1 < num2) {
+      return false;
+    }
+  }
+
+  return true; // Versions are equal
+};
+
+/**
+ * Checks if the multichain accounts feature is enabled for a given state and feature version.
+ *
+ * @param state - The MetaMask state object
+ * @param featureVersion - The specific feature version to check
+ * @returns boolean - True if the feature is enabled for the given state and version, false otherwise.
+ */
+const isMultichainAccountsFeatureEnabled = (
+  state: RemoteFeatureFlagsState,
+  featureVersion: string,
+) => {
```

Comment:
We should probably use `createSelector` or `createDeepEqualSelector` here to memoize and only react to `remoteFeatureFlag` change.

I guess we're gonna use that selector in several place, so memoizing does seem important to me.

We could maybe follow the same pattern than the bridge team: https://github.com/MetaMask/metamask-extension/blob/main/ui/ducks/bridge/selectors.ts#L138-L146? That sounds like a good pattern to me.